File: SwiftProject/SwiftProject/K11.swift
//
//  K11.swift
//  SwiftProject
//
//  Created by 周哲輝 on 2019/05/22.
//  Copyright © 2019 周哲輝. All rights reserved.
//

import Foundation

class K11 : IDispatchable {
    func Question1() {
        var ages = [Int]()
        let MAX_PEOPLE = 5
        
        for i in 1...MAX_PEOPLE {
            print("\(i)人目の年齢を入力してください＞", terminator: "")
            ages.append(Int(readLine()!)!)
        }
        
        let average = ages.reduce(0, +) / ages.count
        print("\(ages.count)人の平均年齢は\(average)です。")
    }
    
    func Question2() {
        print("１つ目の文字列を入力してください。＞", terminator: "")
        var str1 = readLine()!
        print("２つ目の文字列を入力してください。＞", terminator: "")
        let str2 = readLine()!
        str1 += str2
        print("\(str1)の文字数は\(str1.count)です。")
    }
    
    func Question3() {
        var chr: Character
        print("文字を入力してください。＞", terminator: "")
        chr = readLine()!.first!
        var applied = false
        
        if chr.isNumber {
            print("この文字は数字です。")
            applied = true
        }
        if chr.isUppercase {
            print("この文字は英大文字です。")
            applied = true
        }
        if chr.isLowercase {
            print("この文字は英小文字です。")
            applied = true
        }
        if chr.isHexDigit {
            print("この文字は16進数として使える文字です。")
            applied = true
        }
        if chr.isPunctuation {
            print("この文字は区切り文字です。")
            applied = true
        }
        if !applied {
            print("この文字はよくわからない文字です。")
        }
    }
    
    func Question4() {
        let alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        let inputString = "I am a STUDENT. I am good at GAME PROGRAMMING."
        var count = 0
        var _count = 0
        
        print("文字列：\(inputString)")
        print("\n文字頻度")
        for s in alphabets {
            for _s in inputString {
                if String(s) == _s.uppercased() {
                    count += 1
                }
            }
            print("\(s): \(count)\t", terminator: "")
            count = 0
            _count += 1
            if _count % 4 == 0 {
                print()
            }
        }
        print()
    }
    
    
}

File: Java/src/com/ecc/gkgo1a/K11.java
package com.ecc.gkgo1a;

import java.util.Scanner;
import java.util.Vector;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class K11 extends IExecutable {

    @Override
    public void Question1() {
        Vector<Integer> ages = new Vector<>();

        Scanner in = new Scanner(System.in);
        int MAX_PEOPLE = 5;
        for (int i = 0; i < MAX_PEOPLE; i++)
        {
            System.out.print((i + 1) + "人目の年齢を入力して下さい：");
            ages.add(in.nextInt());
        }

        var _average = ages.stream().mapToDouble(x -> x).average();
        double average = 0;
        if (_average.isPresent())
        {
            average = _average.getAsDouble();
        }

        System.out.println(String.format("%d人の平均年齢は%.1fです。", ages.size(), average));
    }

    @Override
    public void Question2() {
        System.out.print("1つ目の文字列を入力して下さい。＞");
        Scanner reader = new Scanner(System.in);
        String str1 = reader.nextLine();
        System.out.print("2つ目の文字列を入力して下さい。＞");
        String str2 = reader.nextLine();
        StringBuilder sb = new StringBuilder(str1);
        sb.append(str2);
        System.out.println(String.format("%sの文字数は%dです。", sb.toString(), sb.toString().length()));
    }

    @Override
    public void Question3() {
        char chr;
        System.out.print("文字を入力して下さい。＞");
        Scanner reader = new Scanner(System.in);
        chr = reader.nextLine().charAt(0);
        boolean applied = false;

        if (Character.isDigit(chr))
        {
            System.out.println("この文字は数字です。");
            applied = true;
        }
        if (Character.isUpperCase(chr))
        {
            System.out.println("この文字は英大文字です。");
            applied = true;
        }
        if (Character.isLowerCase(chr))
        {
            System.out.println("この文字は英小文字です。");
            applied = true;
        }
        Pattern pattern = Pattern.compile("^[A-Fa-f0-9]$");
        Matcher matcher = pattern.matcher(Character.toString(chr));
        if (matcher.matches())
        {
            System.out.println("この文字は16進数として使える文字です。");
            applied = true;
        }

        pattern = Pattern.compile("^(.|,|!|\\?|:|;|\"|'|\\|-|\\(|\\)])$");
        matcher = pattern.matcher(Character.toString(chr));
        if (matcher.matches())
        {
            System.out.println("この文字は区切り文字です。");
            applied = true;
        }

        if (!applied)
            System.out.println("この文字はよくわからない文字です。");
    }

    @Override
    public void Question4() {
        String alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        String inputString = "I am a STUDENT. I am good at GAME PROGRAMMING.";
        int count = 0;
        int _count = 0;
        System.out.println(String.format("文字列：%s\n", inputString));
        System.out.println("文字頻度");
        for (var s :
                alphabets.toCharArray()) {
            for (var _s :
                    inputString.toCharArray()) {
                if (s == Character.toUpperCase(_s)) {
                    count++;
                }
            }
            System.out.print(String.format("%c: %d\t", s, count));
            count = 0;
            _count++;
            if (_count % 4 == 0) {
                System.out.println();
            }
        }
    }
}
